test/preflight/fly_deploy_test.go

```diff
@@ -323,3 +323,14 @@ func TestFlyDeploy_CreateBuilderWDeployToken(t *testing.T) {
 	f.OverrideAuthAccessToken(f.Fly("tokens deploy").StdOutString())
 	f.Fly("deploy")
 }
+
+func TestDeployManifest(t *testing.T) {
+	f := testlib.NewTestEnvFromEnv(t)
+
+	appName := f.CreateRandomAppName()
+	f.Fly("launch --org %s --name %s --region %s --image nginx --internal-port 80 --ha=false", f.OrgSlug(), appName, f.PrimaryRegion())
+
+	f.Fly("deploy --export-manifest manifest.json")
```

Comment:
I've also manually tested the two commands above!

- [ ] deploy --export-manifest manifest.json **creates a manifest.json file:**
    >Running `flyctl.exe deploy --export-manifest manifest.json`, provides the following relevant results:
       > The "Deploy manifest saved to manifest.json" log was printed
       > Created a manifest.json file in the root directory of my Laravel Fly App project
 
- [ ] deploy --manifest manifest.json, **will deploy using the manifest.json file,** and **reflect changes to the machine**
     >Now the fun part, to see if the next command works( which, I think can modify the deployment process right?), I added a new environment variable in manifest.json:
     > ```
     >   "env": {
     >        "APP_ENV": "production",
     >       "DB_CONNECTION": "pgsql",
     >       "LOG_CHANNEL": "stderr",
     >       "LOG_LEVEL": "info",
     >       "LOG_STDERR_FORMATTER": "Monolog\\Formatter\\JsonFormatter",
     >       "SESSION_DRIVER": "cookie",
     >       "SESSION_SECURE_COOKIE": "true",
     >  +   "test": "test"
     >   },
     >  ```
     > Then I run the command `flyctl.exe deploy --manifest manifest.json`. Afterwards, I connected with the machine with `fly ssh console`, checked if that env variable was reflected, and it was!
 
 
![image](https://github.com/user-attachments/assets/9c68e676-3129-458d-863b-7150a4055eb7)

Reply:
@KTanAug21  Yeah, that's exactly it. The manifest can be sent to a remote machine that can resume the deployment without uploading the whole codebase. I hadn't thought that you could modify the manifest post-generating it. This makes me wonder if that's desirable or if we'd want to make it immutable by generating some signature that will tell us if it was modified after it was generated.

Reply:
I see! Is this behavior of the manifest "being modifiable" the reason, why you've added that case "-" for g[etting the manifest from  `io.In`?](https://github.com/superfly/flyctl/pull/3769/files#diff-f1ed021d538250bb75f4922951ea5c467ce53115b46483d10c198a5c88c9cc60R282).

Will getting the manifest stream from io.In make sure that the manifest content won't be modifiable?